graphs/growth.py

```diff
@@ -36,55 +35,82 @@ def TG_formula(TG, A):
+                self.active = [point['positive'] - point['recovered'] - point['dead'] for point in data]
+                self.dead = [point['dead'] for point in data]
+                self.recovered = [point['recovered'] for point in data]
+                self.date_list = [point['date'] for point in data]
+            except yaml.YAMLError as exc:
+                raise exc
+
+        self.cumulative_active = list(filter(lambda x: x >= self.country_basic.case_count, itertools.accumulate(self.active)))
+        self.date_list = self.date_list[len(self.active) - len(self.cumulative_active):]
+        self.x = np.arange(1, len(self.cumulative_active) * 2)
+        self.y = self.country_basic.formula.lambd(self.x)
+
+        self.last_date = datetime.strptime(self.date_list[-1], '%Y-%m-%d')
+        self.date_list += [(self.last_date + timedelta(days=d)).strftime('%Y-%m-%d') for d in range(1, 51)]
+
+    def create_country_figure(self):
+    
+        layout = Layout(title=f"Active cases in {self.country_basic.name}",
+                        xaxis=dict(autorange=True,
+                                title=r'$\text{Days since the 200}^\mathrm{th}\text{ case}$'),
+                        yaxis=dict(type='log', autorange=True, title='COVID-19 active cases'),
+                        hovermode='x',
+                        font={'size': 15},
+                        legend=dict(x=0.01, y=0.99, borderwidth=1))
+
+        figure = Figure(layout=layout)
+        figure.add_trace(
+            Scatter(x=self.x,
+                    y=self.cumulative_active,
+                    mode='lines+markers',
+                    name=f"Active cases",
+                    line={'width': 3},
+                    marker={'size': 8}))
+
+        figure.add_trace(
+            Scatter(
+                x=self.x,
+                y=self.y,
+                text=self.date_list,
+                mode='lines',
+                name=self.country_basic.formula.text,
+                line={
+                    'dash': 'dash',
+                    'width': 3
+                },
+            ))
+        return figure
+
+if args.country != "ALL":
+    country = next(c for c in countries if c.name == args.country)
+    country_data = CountryData(args.data, args.country, country)
+    country_data.create_country_figure().show()
+    # offline.plot(country_data.create_country_figure(), filename='graph.html')
+    exit(0)
+
+countries_data = [CountryData(args.data, country.name, country) for country in countries]
+
+import dash
+import dash_core_components as dcc
+import dash_html_components as html
+
+app = dash.Dash(external_scripts = [
+        'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML'
+    ])
+
+graphs = [dcc.Graph(id = country_data.name, figure = country_data.create_country_figure())
+            for country_data in countries_data]
+
+app.layout = html.Div(graphs)
+app.run_server(port=8080)
```

Comment:
Spojenie treba otvoriť vonkajšiemu svetu:
```suggestion
app.run_server(host="0.0.0.0", port=8080)
```

Reply:
Done